src/main/java/net/dv8tion/jda/api/interactions/commands/build/CommandData.java

```diff
@@ -61,10 +66,45 @@ public interface CommandData extends SerializableData
      *         True, if this command is enabled by default for everyone. (Default: true)
      *
      * @return The builder instance, for chaining
+     *
+     * @deprecated This has been deprecated in favor of {@link #setDefaultPermissions(CommandPermissions)}
      */
     @Nonnull
+    @Deprecated
+    @ForRemoval
+    @DeprecatedSince("5.0.0")
+    @ReplaceWith("CommandData#setDefaultPermissions(CommandPermissions)")
     CommandData setDefaultEnabled(boolean enabled);
 
+    /**
+     * Sets the default {@link CommandPermissions} for this command.
+     * <br>By default, everyone can use this command. This can be configured or overwritten by moderators in each guild.
+     *
+     * <p>Passing {@link CommandPermissions#DISABLED} will only show this command to admins.
+     *
+     * @param  permission
+     *         {@link CommandPermissions} representing the default permissions of this command.
+     *
+     * @throws IllegalArgumentException
+     *         If any of the provided Permissions are null or {@link Permission#UNKNOWN UNKNOWN}.
+     *
+     * @return The builder instance, for chaining
+     */
+    @Nonnull
+    CommandData setDefaultPermissions(@Nonnull CommandPermissions permission);
+
+    /**
+     * Sets whether this command is only usable in a guild.
+     * <br>This only has an effect if this command is registered globally.
+     *
+     * @param  guildOnly
+     *         Whether to restrict this command to guilds
+     *
+     * @return The builder instance, for chaining
+     */
+    @Nonnull
+    CommandData setGuildOnly(boolean guildOnly);
```

Comment:
Specify that this has no use for guild commands, and say what the default behavior is